src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -1337,7 +1337,7 @@ protected void GenerateFindFirstChar()
                 // ch = runtext[runtextpos];
                 // if (ch == lastChar) goto partialMatch;
```

Comment:
Thanks for fixing this.

Can you move this second line of the comment down to line 1352? The problem here is that this comment suggests the casing is only relevant to that one comparison, but we actually store it into a local and use it for multiple comparisons, and not just against this last character being checked here.